scalameta/src/main/scala/scala/meta/syntactic/Token.scala

```diff
@@ -153,3 +152,88 @@ object Token {
     def end = content.chars.length
   }
 }
+
+// NOTE: We have this unrelated code here, because of how materializeAdt works.
+//
+// Due to an unfortunate limitation of knownDirectSubclasses,
+// all macro applications that depend on that API
+// must come after the classes that they call the API on.
+//
+// That's why if we put TokenLiftables elsewhere, we might run into troubles
+// depending on how the file and its enclosing directories are called.
+// To combat that, we have TokenLiftables right here, guaranteeing that there won't be problems
+// if someone wants to refactor/rename something later.
+trait TokenLiftables extends tokens.Liftables {
```

Comment:
Yes, that's right. Ideally, we shouldn't have this `customMatcher` at all and the materializer should simply account for the implicits in scope. In the future, I'll get to implementing that.